R/glimpse.R

```diff
@@ -128,7 +128,11 @@ format_glimpse <- function(x, ...) {
 
 # A variant with collapse and without checks, for format_glimpse.list()
 format_glimpse_1 <- function(x, ...) {
-  collapse(format_glimpse_(x, ...))
+  if (!is_null(list2(...)[[".inner"]]) && list2(...)[[".inner"]]) {
```

Comment:
Instead of doing this, can we add `.inner = NULL` to the signature and check if it's `NULL` ?

Reply:
good catch, done